CT-2318 Redirect if case is deleted

Polsonby opened the pull request:
## Description
One of our most frequently recurring Sentry issues - it transpired that the issue happens when someone clicks a link to edit the case assignment from an email but on a case that has been subsequently deleted. The exception is caused most directly because we were checking for the existence of the assignment directly via Assignment, but then trying to reach it through the case which is nil at that point. 

The wider problem is that the system was attempting to display the edit assignment page in the first place on a deleted case, so I added a before action to redirect to the home page with an appropriate message, and a feature spec to reproduce the bug and validate the new behaviour. 
 
## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [x] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
 
### Screenshots
<!-- Screenshots of the new changes if appropriate -->
![image](https://user-images.githubusercontent.com/1161161/64417180-84007000-d090-11e9-98a2-65802404ac8c.png)

 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2318
 
### Deployment
none
 
### Manual testing instructions
none

Polsonby committed:
CT-2318 Redirect if case is deleted

mseedat-moj reviewed (changes requested):
See git patch file for changes to the test and some ideas for changing `assignments_controller`

https://drive.google.com/open?id=1m_S-j2n1zcIFGp78bkj2h6Ld0zNo2vNU

Polsonby committed:
CT-2318 Rename method and apply to other actions

It's likely that the same problem exists for other actions triggered on
cases that have since been deleted, so moved the check into the existing
before_action handler

Polsonby committed:
CT-2318 Refactor

mseedat-moj reviewed (approved):
Excellent sleuthing

Polsonby committed:
Trigger rebuild
